packages/maker/src/api/defineInterface.ts

```diff
@@ -20,20 +20,36 @@ import { defineSharedPropertyType } from "./defineSpt.js";
 import type { BlueprintIcon } from "./iconNames.js";
 import type {
   InterfaceType,
+  InterfaceTypeRid,
+  InterfaceTypeStatus,
+  InterfaceTypeStatus_active,
+  InterfaceTypeStatus_experimental,
   PropertyTypeType,
   SharedPropertyType,
 } from "./types.js";
 
+/**
+ * This status indicates that the interface is reaching the end of its life and will be removed as per the
+ * deadline specified.
+ */
+export interface DeprecatedInterfaceTypeStatus {
+  message: string;
+  deadline: string;
+  replacedBy: InterfaceTypeRid | undefined;
```

Comment:
Generally not a fan of rids inside OAC. I believe this won't even work under the hood, because we removed all Rids and replaced them with API names. Can we remove this for now given its optional?